enhancements/cluster-logging/elasticsearch-cert-management.md

```diff
@@ -0,0 +1,225 @@
+- [ ] Test plan is defined
+- [ ] Migration plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Currently to use the Elasticsearch Operator, one would need to generate secrets in a specific manner which creates a high amount of complexity for use.
+This proposal seeks to outline a mechanism where the Elasticsearch Operator creates and maintains these certificates and the elasticsearch/kibana secret instead of other operators (e.g. Cluster Logging and Jaeger), and allows an annotation mechanism on secrets for injecting required keys and certificates for mTLS with Elasticsearch.
+
+## Motivation
+
+Currently our managed Elasticsearch cluster requires anything using our operator to generate, manage, and provide certs and secrets for the operator and its cluster to use.
+This leads to many potential sources to create certificates and can cause a gap between what the managed Elasticsearch requires and what something may provide.
+This would instead make the Elasticsearch Operator a single source of truth for how certificates would be generated.
+
+### Goals
+The specific goals of this proposal are:
+
+* Outline the responsibility for the Elasticsearch Operator regarding certificates
+
+* Determine an annotation that the Elasticsearch Operator will watch for on secrets in the same namespace as an existing elasticsearch CR
+
+* Discuss upgrade strategies from the current LTS mechanism.
+
+We will be successful when:
+
+* Users of Elasticsearch Operator no longer need to maintain certificates for Elasticsearch and can have certificates injected into their secrets
+
+* There is a clear path forward for automatic upgrades
+
+### Non-Goals
+
+* This does not seek to implement work to adhere to the cluster level TLS definitions, however this does make the implementation less complex for the Elasticsearch use-case.
+
+### Current workflow vs Proposed
+
+Currently, a consumer like Jaeger or Cluster Logging is responsible for the following for their ES cluster:
+
+1. Generating and managing their own keys/certificates including those that are used in Elasticsearch as server serving and inter-cluster so that Elasticsearch can join and form clusters.
+1. Generating and managing any client keys/certs to communicate with Elasticsearch. This would include their forwarders, and if applicable, their UIs.
+1. Generating and managing secrets to be used by: EO, Elasticsearch, IndexManagement cronjobs, and any clients as denoted in point 2.
+1. Creating and updating their elasticsearch CR(s) so that EO can manage their cluster.
+
+With this proposal the workflow would be:
+
+1. Create their elasticsearch CR with an annotation to denote EO will be managing certs for the cluster
+1. Add an additional annotation on the elasticsearch CR for each client they need to have certs injected into a specific secret.
```

Comment:
do these injected secrets get rotated or expire?

Reply:
since EO would be managing them, they should be rotated. I'll clarify